parking_lot.py

```diff
@@ -0,0 +1,172 @@
+    start: datetime | None
+    end: datetime | None = None
+    fee: float | None = Field(default=None, ge=0, description=f"The fee must be greater or equal to 0.")
+
+    @validator('location')
+    def location_must_be_greater_than_zero(cls, v) -> int:
+        if v < 0:
+            raise ValueError('must be greater than zero')
+        return v
+
+    def get_exit_fee(self, start: datetime, end: datetime, tariff: float) -> float:
+        start_date = start.replace(tzinfo=None)
+        time_spent_when_parked = (end - start_date).total_seconds()
+
+        if time_spent_when_parked < FREE_PARKING_TIME:
+            return 0
+        else:
+            match tariff:
+                case TariffName.hourly:
+                    exit_fee = math.ceil(time_spent_when_parked / SECONDS_IN_HOUR) * Tariffs.hourly
+                    return exit_fee
+                case TariffName.daily:
+                    exit_fee = math.ceil(time_spent_when_parked / SECONDS_IN_DAY) * Tariffs.daily
+                    return exit_fee
+
+
+@app.exception_handler(InvalidLocationException)
+async def invalid_location_exception_handler(exception: InvalidLocationException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Cannot park your car there because location {exception.location} is invalid!"})
+
+
+@app.exception_handler(LocationNotAvailableException)
+async def location_not_available_exception_handler(exception: LocationNotAvailableException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Cannot park your car there because location {exception.location} is not available!"})
+
+
+@app.exception_handler(StartDateException)
+async def start_date_in_the_future(exception: StartDateException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Cannot part your car because date {exception.start} is invalid!"})
+
+
+@app.exception_handler(CarIdExistsException)
+async def car_id_already_exists(exception: CarIdExistsException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Car {exception.car_id} is already parked!"})
+
+
+@app.exception_handler(CarIdDoesNotExistException)
+async def car_id_does_not_exist(exception: CarIdDoesNotExistException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Car {exception.car_id} is not in the parking lot!!"})
+
+
+@app.exception_handler(CarAlreadyRemovedException)
+async def car_id_already_removed(exception: CarAlreadyRemovedException):
+    return JSONResponse(status_code=422, content={"status": "error", "message": f"Car {exception.car_id} already removed!"})
+
+
+@app.get("/cars/")
+async def get_cars():
+    return PARKED_CARS
+
+
+@app.get("/cars/{car_id}", status_code=status.HTTP_200_OK)
+async def get_car(car_id: Optional[str] = None):
+    return PARKED_CARS[car_id]
```

Comment:
I think plate names like: "!@#!" or "'"" and similar are accepted, while not restricted in requirements, could be a problem since they aren't known valid identifiers

Reply:
Must be an issue in the requirements. Added validator.